packages/langium/src/workspace/document-builder.ts

```diff
@@ -4,19 +4,19 @@
  * terms of the MIT License, which is available in the project root.
  ******************************************************************************/
 
-import type { URI } from '../utils/uri-utils.js';
+import { CancellationToken, Disposable } from 'vscode-jsonrpc';
```

Comment:
Suggestion: The `vscode-jsonrpc` library is roughly 50kb in size. Seems like quite a lot if we only need to import some utilities from it. We should implement these ourselves. We actually already have a `Disposable` implementation on our own.

Reply:
Wouldn't tree-shaking drop all the unused overhead?

Reply:
`vscode-jsonrpc` is a commonjs package, tree-shaking usually doesn't work on those.

Reply:
I'm not sure how well bundlers can optimize commonjs packages. See [here](https://web.dev/articles/commonjs-larger-bundles) for example.